src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -35,28 +35,29 @@ import styles from '../../../styles/styles';
 import * as SessionUtils from '../../SessionUtils';
 import getNavigationModalCardStyle from '../../../styles/getNavigationModalCardStyles';
 
-let currentUserEmail;
+let timezone;
+let currentAccountID;
 Onyx.connect({
     key: ONYXKEYS.SESSION,
     callback: (val) => {
-        // When signed out, val is undefined
-        if (!val) {
+        // When signed out, val hasn't accountID
+        if (!_.has(val, 'accountID')) {
+            timezone = null;
```

Comment:
Coming from https://github.com/Expensify/App/issues/48941, we missed that the `currentAccountID` isn't actually cleared in the case a user deletes their account. This can cause a loop of trying to update the timezone which never resolves because the update timezone call's error (due to the account no longer existing) isn't handled.